### PR TITLE
Add upgrade lock file and healthcheck to the container

### PR DESCRIPTION
**Pull request by justinclift**
This is just a copy of #60, but in a branch of the main repo so we can see the CI check results. :smile:

### CODE REVIEW

**Comment by justinclift** [comment]
k, that looks happier. :smile:

Will need to properly look over the code, which I'll try to get done later today. :smile:

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -2,6 +2,9 @@
 set -Eeo pipefail
 # TODO swap to -Eeuo pipefail above (after handling all potentially-unset variables)
 
+# Define the path to the upgrade lock file using PGDATA if set, otherwise default
+UPGRADE_LOCK_FILE="${PGDATA:-/var/lib/postgresql}/upgrade_in_progress.lock"
```
Comment (justinclift) [inline docker-entrypoint.sh:6]: Oh.  I think the default path for `PGDATA` might need `/data` appended on the end there.  Will double check when I look at this properly in a few hours. :smile:

File: pgautoupgrade-healthcheck.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+# Define the path to the upgrade lock file using PGDATA if set, otherwise default
+UPGRADE_LOCK_FILE="${PGDATA:-/var/lib/postgresql}/upgrade_in_progress.lock"
```
Comment (justinclift) [inline pgautoupgrade-healthcheck.sh:4]: Same thought as above. :smile:

File: docker-entrypoint.sh (hunk 1 above)
Comment (justinclift) [reply docker-entrypoint.sh:6]: Yeah. PGDATA should definitely be `/var/lib/postgresql/data` if nothing is passed.

Comment (wuast94) [reply docker-entrypoint.sh:6]: @justinclift will do it later, if im intrestet :P

Comment (wuast94) [reply docker-entrypoint.sh]: later is now xD

Comment (justinclift) [reply docker-entrypoint.sh]: Awesome, thanks. :smile:

**Pull request merged**